Update buildpack spec for RFC-0036 (#99)
buildpacks/spec: buildpack.md

### Provided by the Lifecycle

#### Buildpack Specific Variables

> The following environment variables MUST be set per-buildpack.

Reviewer: ```suggestion
The following environment variables MUST be set by the lifecycle in each buildpack's execution environment.
```

> These variables MAY change between execution of each buildpack in a single lifecycle phase.

Reviewer: ```suggestion
These variables MAY differ between buildpacks.
```

> The following additional environment variables MUST NOT be overridden by the lifecycle.

Reviewer: This table appears to list env vars that are set on stack and should not be modified by the lifecycle. In this case `CNB_BUILDPACK_DIR` would **not** be set on the stack. The lifecycle should actively set it to the correct value for each buildpack. Because there is no existing case like this we may need to make a new section.

Author: yup, agree. i'll fix it.

The following additional environment variables MUST NOT be overridden by the lifecycle.

> | Env Variable        | Description                          | Detect | Build | Launch

Reviewer 2: I think this indent should be pulled back, effectively no-oping this change.

The following additional environment variables MUST NOT be overridden by the lifecycle.

| Env Variable        | Description                          | Detect | Build | Launch
|---------------------|--------------------------------------|--------|-------|--------

> | `CNB_STACK_ID`      | Chosen stack ID                      | [x]    | [x]   |
> | `CNB_BUILDPACK_DIR` | The root of the buildpack source     | [x]    | [x]   |
> | `BP_*`              | User-provided variable for buildpack | [x]    | [x]   |

Reviewer: ```suggestion
| `CNB_STACK_ID`      | Chosen stack ID                      | [x]    | [x]   |
| `BP_*`              | User-provided variable for buildpack | [x]    | [x]   |
```
